src/interpreter/internal.rs

```diff
@@ -166,10 +166,41 @@ impl<S> Interpreter<S> {
         }?;
 
         // update the local copy of the output
-        let mut output = Output::variable(owner_to_set, amount_to_set, asset_id_to_update);
+        let output = Output::variable(owner_to_set, amount_to_set, asset_id_to_update);
+        self.set_output(out_idx, output)
+    }
+
+    pub(crate) fn revert_variable_output(&mut self, out_idx: usize) -> Result<(), RuntimeError> {
+        if out_idx >= self.tx.outputs().len() {
+            return Err(PanicReason::OutputNotFound.into());
+        }
+
+        let reverted_variable_output = Output::variable(Default::default(), 0, Default::default());
+        self.set_output(out_idx, reverted_variable_output)
+    }
+
+    pub(crate) fn set_change_output(&mut self, out_idx: usize, update_amount: Word) -> Result<(), RuntimeError> {
+        let mut output = *self.tx.outputs().get(out_idx).ok_or(PanicReason::OutputNotFound)?;
+        if let Output::Change { amount, .. } = &mut output {
+            *amount = update_amount;
+        } else {
+            // TODO: make a panic reason for ExpectedChangeOutput
+            return Err(PanicReason::OutputNotFound.into());
```

Comment:
Ditto to above, the VM itself doesn't set change, that's technically done _after_ VM execution.